temba/api/v2/views.py

```diff
@@ -3462,3 +3464,69 @@ def get_read_explorer(cls):
             "params": [],
             "example": {},
         }
+
+
+class ValidateUrlAttachmentEndpoint(BaseAPIView):
+    """
+    This url allows you validate attachment url.
+
+    POST /api/v2/validate_attachment_url
+
+    * **attachment_url** - the url of attachment
+
+    Response:
+
+        # attachment valid
+        {
+            "valid": true,
+            "type": "image",
+            "size": 9604
+        }
+
+        # attachment invalid
+        {
+            "valid": false,
+            "error": "Invalid Attachment. Attachments must be either video, audio, or an image."
+        }
+    """
+
+    def post(self, request, *args, **kwargs):
+        status_code = 200
+        validation_data = {}
+        serializer = UrlAttachmentValidationSerializer(data=request.data)
+
+        if serializer.is_valid():
+            url = serializer.validated_data.get("attachment_url")
+            response = requests.head(url)
+            file_type = response.headers.get("content-type", "").strip()
+            file_type = file_type.split("/")[0] if file_type else ""
+            file_size = response.headers.get("content-length")
+            file_size = int(file_size) if file_size else None
+            status_code = response.status_code
+
+            if status_code != 200:
+                validation_data.update({"valid": False, "error": _("Url of attachment is not valid.")})
+
+            elif file_type not in ("image", "audio", "video"):
```

Comment:
For sending an email with attachments, I think we should accept all files except this list of extensions: https://github.com/greatnonprofits-nfp/floweditor/pull/38/files#diff-f88ed0b807262cd475ff365a82ce8120L30, because .pdf, .docx, .xlsx should be acceptable as well at this flow step.

Reply:
OK. I can, probably, convert content-type or the URL to file extension. And validate by the extension.